### ISSUE TITLE
Make getFilterJsonSchemaFor() model-aware

### ISSUE THREAD

**Issue by achrinza** [strongloop/loopback-next#4977]
Currently, [`getFIlterJsonSchemaFor()` isn't model-aware](https://github.com/strongloop/loopback-next/blob/85c12eb2ce12dfd54cc70d28d3d39e17abfe0142/packages/repository-json-schema/src/filter-json-schema.ts#L54-L55).

This means that the `openapi.json` order filter parameter will be generic:

![image](https://user-images.githubusercontent.com/25147899/77757246-d5dc0800-706b-11ea-90d3-6d5937315eb3.png)

In the case above, it's the `/todos` parameter from the `Todo` example.

This would lead to issues where the suggested `order` filter isn't a valid query:

![image](https://user-images.githubusercontent.com/25147899/77756908-1d15c900-706b-11ea-847b-15871211d188.png)

(Notice the `"order": [ "string" ]`)

## Steps to reproduce

A quick way to reproduce is use the Todo example:

`lb4 example todo`

This also applies to any LoopBack 4 application that utilises `@param.filter(ModelHere)`.

## Expected Behavior

The `order` filter should be model-aware and not generic.

## Link to reproduction sandbox

`lb4 example todo`

## Related Issues

_See [Reporting Issues](http://loopback.io/doc/en/contrib/Reporting-issues.html) for more tips on writing good issues_

**Comment by garethhcoleman** [comment]
PR #4978 adds a line to todos tutorial to aid new users who encounter this behaviour whilst underlying feature/bug is developed/resolved.

**Comment by emonddr** [comment]
@achrinza , apologies, but I don't follow. The Todo examples don't involve any Order object that I am aware of. To help investigate the issue, please provide a minimal app demonstrating the problem. Thanks.

**Comment by achrinza** [comment]
@emonddr, apologies if I wasn't clear - what I meant was the `order`  **filter** that is passed to repository functions.

This is added to the OpenAPI spec by `@param.filter()` from `@loopback/rest` which eventually calls `getFilterJsonSchemaFor()` which isn't model-aware, thereby leading to  the API Explorer generating an unusable default `filter` parameter.

I'll see if I can create a minimal app later today.

**Comment by agnes512** [comment]
I was checking filters these days. `order` filter works well with API Explorer when I tested it.
Both work:
```
{
  "order": [
    "number ASC"
  ]
}
or
{
  "order": [
    "number ASC", "name DESC"
  ]
}
```
The generated default parameter makes sense to me.  Could you explain more about the issue?

**Comment by achrinza** [comment]
@agnes512 Currently, the default parameter generated by the API Explorer is:

```json
{
  "order": [
    "string"
  ]
}
```

While it does makes sense, it can't be used as-is as `"string"` will need to be replaced with either `ASC`, `DESC` or `propertyName ASC`, etc.

Since the [API docs for `getFilterJsonSchema()`](https://github.com/strongloop/loopback-next/blob/85c12eb2ce12dfd54cc70d28d3d39e17abfe0142/packages/repository-json-schema/src/filter-json-schema.ts#L54-L55) suggests that this was a future improvement that was considered before, it may be a good opportunity to revisit it and see if we can improve the developers' experience with using API Explorer.

This is improvement is especially useful since the default parameter generated by the API Explorer is rather cryptic for developers who are not familiar with the valid `order` filter structures.

**Comment by agnes512** [comment]
Agree, `"order ASC"` is not intuitive enough. Maybe we can have a layer to convert  `"order": "ASC"` to `"order ASC"`, thoughts? @raymondfeng

**Comment by bajtos** [comment]
A possible quick fix is to add an empty array as an example value for `order`. We can change the following block:

https://github.com/strongloop/loopback-next/blob/9478d319a472e1dca4122a29dbb414ce1525bdb0/packages/repository-json-schema/src/filter-json-schema.ts#L87-L92

as follows:

```ts
    order: {
      type: 'array',
      items: {
        type: 'string',
      },
      examples: [[]],
    },
```

That gives me the following default value in REST API Explorer:

```json
{
  "offset": 0,
  "limit": 100,
  "skip": 0,
  "order": [],
  "where": {
    "additionalProp1": {}
  },
  "fields": {
    "id": true,
    "title": true,
    "desc": true,
    "isComplete": true,
    "remindAtAddress": true,
    "remindAtGeo": true,
    "tag": true
  }
}
```

While not ideal from the point of user/developer experience, it's at least a valid filter.

**Comment by bajtos** [comment]
Having said that, I am not a big fan of that large default filter value, and personally would prefer a much smaller one.

When I modify https://github.com/strongloop/loopback-next/blob/9478d319a472e1dca4122a29dbb414ce1525bdb0/packages/repository-json-schema/src/filter-json-schema.ts#L107-L113

as follows

```ts
  const schema: JsonSchema = {
    ...(options.setTitle !== false && {
      title: `${modelCtor.modelName}.Filter`,
    }),
    properties,
    additionalProperties: false,
    examples: [{limit: 100}],
  };
```

then the REST API Explorer offers me the following nice default:

```json
{
  "limit": 100
}
```
